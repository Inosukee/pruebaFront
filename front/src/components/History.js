import React from 'react'

export const History = () => {
    return (
        <div className='history__content'>
            <ul className='history__history'>
                {/* <li>
                    <a href="#">prueba</a>
                    <div className='history__svg'>
                        <svg xmlns="http://www.w3.org/2000/svg" width="6" height="8"><path fill="none" stroke="#666" d="M1 0l4 4-4 4"></path></svg>
                    </div>
                </li>
                <li>
                    <a href="#">prueba</a>
                    <div className='history__svg'>
                        <svg xmlns="http://www.w3.org/2000/svg" width="6" height="8"><path fill="none" stroke="#666" d="M1 0l4 4-4 4"></path></svg>
                    </div>
                </li>
                <li>
                    <a href="#">prueba</a>
                </li> */}
            </ul>
        </div>
    )
}
