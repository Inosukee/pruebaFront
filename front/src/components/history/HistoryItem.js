import React from 'react'

export const HistoryItem = ({ name, last = false }) => {
  return (

    !last ?
      (
        <li>
          <a href="!#">{name}</a>
          <div className='history__svg'>
            <svg xmlns="http://www.w3.org/2000/svg" width="6" height="8"><path fill="none" stroke="#666" d="M1 0l4 4-4 4"></path></svg>
          </div>
        </li>

      )
      :
      (
        <li>
          <a href="!#">{name}</a>
        </li>
      )
  )
}
