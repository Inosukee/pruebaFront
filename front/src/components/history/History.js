import React from 'react'
import { HistoryItem } from './HistoryItem'

export const History = ({ categories }) => {
    return (
        <div className='history__content'>
            <ul className='history__history'>
                {
                    categories &&
                    categories.map((history, i, { length }) => {
                        if (i < length - 1) {
                            return (<HistoryItem key={history.id}
                                {...history} />)
                        } else {
                            return (<HistoryItem last={true} key={history.id}
                                {...history} />)
                        }
                    })
                }
            </ul>
        </div>
    )
}
