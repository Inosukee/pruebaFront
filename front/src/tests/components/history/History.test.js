import React from 'react'
import { shallow } from 'enzyme';
import { History } from '../../../components/history/History';


const categoriesTest = [
  { id: 1, name: 'prueba 1' },
  { id: 2, name: 'prueba 2' },
  { id: 3, name: 'prueba 3' }
]

describe('Pruebas en <History />', () => {
  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<History categories={categoriesTest} />)
    expect(wrapper.find('.history__history').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})