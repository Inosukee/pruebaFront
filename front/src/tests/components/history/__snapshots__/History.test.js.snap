// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <History /> debe mostrar el componente correctamente 1`] = `
<div
  className="history__content"
>
  <ul
    className="history__history"
  >
    <HistoryItem
      id={1}
      key="1"
      name="prueba 1"
    />
    <HistoryItem
      id={2}
      key="2"
      name="prueba 2"
    />
    <HistoryItem
      id={3}
      key="3"
      last={true}
      name="prueba 3"
    />
  </ul>
</div>
`;
